### PR TITLE
Parallelize acceptance tests

### PR DESCRIPTION
**Pull request by peterellisjones**
The acceptance tests currently take around 50 minutes to run.

The first 10 minutes of this is deploying the BOSH director using BOSH lite. This can't be easily parallelized.

The next 40 minutes is the tests themselves where we deploy HAProxy to the BOSH lite director in different configurations. This PR parallelizes this process via the `ginkgo -p` [flag](https://onsi.github.io/ginkgo/#parallel-specs) which by default uses one thread per CPU when run on a machine with four or fewer CPUs, or CPUs - 1 when run on a machine with more than four CPUs. It uses seven threads on our Concourse CI server. Note that the HAProxy deployments now need unique names per thread (although deployments in the same thread can reuse the deployment name).

This change reduces the test execution time from 40 minutes to about 20 minutes, so total time is approximately 30 minutes including the initial BOSH director setup.

Note that when deploying a new deployment the bosh director will compile any releases that haven't been compiled before with the required stemcells. To avoid all threads in the parallelized tests running the same compilations at the same time, this PR deploys HAProxy once in a `SynchronizedBeforeSuite` block which runs before any tests. This ensures that the first deployment in each of the test threads can make used of already-compiled releases.

### CODE REVIEW

File: ci/release_notes.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# Fixes
+- acceptance tests are now run in parallel
+
+# Acknowledgements
+
+Thanks @peterellisjones for the PR
```
Comment (maxmoehl) [inline ci/release_notes.md:6]: Do we need a release for this? We don't change anything in the functionality, therefore I wouldn't create one.

File: acceptance-tests/acceptance_tests_suite_test.go [hunk 2]
```diff
@@ -142,3 +169,9 @@ func checkTLSErr(err error, expectString string) {
 	Expect(errors.As(tlsErr, &opErr)).To(BeTrue())
 	Expect(opErr.Err.Error()).To(ContainSubstring(expectString))
 }
+
+func println(s string) {
+	for _, line := range strings.Split(s, "\n") {
+		fmt.Printf("node %d/%d: %s\n", ginkgoConfig.GinkgoConfig.ParallelNode, ginkgoConfig.GinkgoConfig.ParallelTotal, line)
+	}
+}
```
Comment (maxmoehl) [inline acceptance-tests/acceptance_tests_suite_test.go:177]: This shadows the builtin function `println(args ...Type)`. What about just `log(s string)`?

File: ci/release_notes.md (hunk 1 above)
Comment (maxmoehl) [inline ci/release_notes.md:6]: Do we need a release for this? We don't touch any of the actual coding therefore I wouldn't create one.

File: ci/scripts/acceptance-tests [hunk 3]
```diff
@@ -41,6 +41,10 @@ bosh -n upload-stemcell $stemcell_xenial_path
 echo "----- Uploading BPM"
 bosh -n upload-release $bpm_release_path
 
+echo "----- Uploading os-conf (used for tests only)"
+bosh -n upload-release --sha1 386293038ae3d00813eaa475b4acf63f8da226ef \
+  https://bosh.io/d/github.com/cloudfoundry/os-conf-release?v=22.1.2
+
```
Comment (maxmoehl) [inline ci/scripts/acceptance-tests:47]: Why was this added now? I thought this doesn't change any of the tests? And would it be possible to consume the other releases from bosh.io directly as well?

**Review by maxmoehl — CHANGES_REQUESTED** [review]
Looks good overall

File: ci/release_notes.md (hunk 1 above)
Comment (peterellisjones) [reply ci/release_notes.md:6]: Good point

File: acceptance-tests/acceptance_tests_suite_test.go (hunk 2 above)
Comment (peterellisjones) [reply acceptance-tests/acceptance_tests_suite_test.go:177]: We can't use `log` because that is a package in the stdlib which is imported in eg acceptance-tests/remote.go

Comment (peterellisjones) [reply acceptance-tests/acceptance_tests_suite_test.go]: The stdlib function is fmt.Println FYI so it's not shadowed unless you import `fmt` as a dot import (which we're not doing anywhere)

I've changed the function name to `print` to avoid confusion with `fmt.Println`

File: ci/scripts/acceptance-tests (hunk 3 above)
Comment (peterellisjones) [reply ci/scripts/acceptance-tests:47]: Just a small refactor so that all releases are uploaded in the same place before running the tests. This was previously being added via a URL in https://github.com/cloudfoundry/haproxy-boshrelease/blob/master/acceptance-tests/bosh_helpers.go#L50-L57 

>  And would it be possible to consume the other releases from bosh.io directly as well?
yes you could consume the BPM release directly from bosh.io however since BPM is also included in the final bosh release tarball, doing it via the pipeline makes it easier to track which version we have

File: ci/release_notes.md (hunk 1 above)
Comment (peterellisjones) [reply ci/release_notes.md]: good point, removed

File: acceptance-tests/acceptance_tests_suite_test.go (hunk 2 above)
Comment (maxmoehl) [reply acceptance-tests/acceptance_tests_suite_test.go]: I'm talking about the builtin function, not the stdlib. You can call this without any package prefix and you can shadow them by defining something else with the same name. But IMO that's confusing and should be avoided. See https://pkg.go.dev/builtin#println (print is also a builtin).

Comment (peterellisjones) [reply acceptance-tests/acceptance_tests_suite_test.go]: Ah never even heard of that.

I've changed it to `writeLog`

**Review by maxmoehl — APPROVED** [review]
LGTM
